src/webgl/p5.RendererGL.js

```diff
@@ -1724,52 +1726,26 @@ p5.RendererGL = class RendererGL extends Renderer {
   }
 
   /*
-   * selects which fill shader should be used based on renderer state,
-   * for use with begin/endShape and immediate vertex mode.
+   * This method will handle both image shaders and
+   * fill shaders, returning the appropriate shader
+   * depending on the current context (image or shape).
    */
-  _getImmediateFillShader() {
-    const fill = this.userFillShader;
-    if (this._useNormalMaterial) {
-      if (!fill || !fill.isNormalShader()) {
-        return this._getNormalShader();
-      }
+   _getFillShader() {
+    if (this._drawingImage) {
+      return this.userImageShader;
     }
-    if (this._enableLighting) {
-      if (!fill || !fill.isLightShader()) {
-        return this._getLightShader();
-      }
-    } else if (this._tex) {
-      if (!fill || !fill.isTextureShader()) {
-        return this._getLightShader();
-      }
-    } else if (!fill /*|| !fill.isColorShader()*/) {
-      return this._getImmediateModeShader();
+    else if (this.userFillShader) {
+      return this.userFillShader;
     }
-    return fill;
-  }
-
-  /*
-   * selects which fill shader should be used based on renderer state
-   * for retained mode.
-   */
-  _getRetainedFillShader() {
-    if (this._useNormalMaterial) {
+    else if (this._useNormalMaterial) {
       return this._getNormalShader();
     }
-
-    const fill = this.userFillShader;
-    if (this._enableLighting) {
-      if (!fill || !fill.isLightShader()) {
-        return this._getLightShader();
-      }
-    } else if (this._tex) {
-      if (!fill || !fill.isTextureShader()) {
-        return this._getLightShader();
-      }
-    } else if (!fill /* || !fill.isColorShader()*/) {
+    else if (this._enableLighting || this._tex) {
+      return this._getLightShader();
+    }
+    else {
```

Comment:
could probably remove this else block, it's obvious that it could return `this._getColorShader();` .

Reply:
we could just write return `this._getColorShader();` without the else block.